ShoppingM8-be/src/main/java/edu/pw/shoppingm8/list/ListServiceImpl.java

```diff
@@ -69,4 +67,9 @@ public void checkIfUserIsOwner(List list) {
     public Collection<List> getUsersLists(User user) {
         return listRepository.findByOwner(user);
     }
+
+    @Override
+    public void checkIfListExistsAndUserHasAccess(Long listId) {
```

Comment:
List not being in the database can be considered an exceptional situation. Maybe it's not necessary to create another method for cheking access that does exactly the same when situation of not existing is handled by exception anyway.

Reply:
It is for a situation where retrieving a list is not required (retrieving a product), to provide a specific exception that results in a correct http status code and gives message indicating what happened. However if you believe it would be better to avoid such check I will change it.

Reply:
Ok, sorry, I didn't notice that "checkIfUserHasAccess" does not retrieve the list. Let it stay the way it is.